app/classifier/tasks/dropdown/index.jsx

```diff
@@ -0,0 +1,292 @@
+            displayAnswer = getOptionLabel(answer.value, i);
+          } else {
+            displayAnswer = answer.value;
+          }
+          return (<div key={i} className="answer">
+            <i className="fa fa-arrow-circle-o-right" /> {props.task.selects[i].title} - {displayAnswer}
+          </div>);
+        })}
+      </div>
+    </div>);
+};
+
+DropdownSummary.propTypes = {
+  task: PropTypes.shape({
+    instruction: PropTypes.string,
+    selects: PropTypes.array
+  }),
+  annotation: PropTypes.shape({
+    value: PropTypes.array
+  })
+};
+
+DropdownSummary.defaultProps = {
+  task: {},
+  annotation: {}
+};
+
+export default class DropdownTask extends React.Component {
+  componentWillMount() {
+    this.menus = [];
+  }
+
+  componentDidMount() {
+    this.setDefaultValues();
+
+    if (this.props.autoFocus === true) {
+      this.handleFocus();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.task !== this.props.task) {
+      this.setDefaultValues();
+      if (this.props.autoFocus === true) {
+        this.handleFocus();
+      }
+    }
+  }
+
+  onChangeSelect(i, option) {
+    let newOption = option;
+    if (!newOption) {
+      newOption = { label: null, value: null };
+    }
+    const { value } = this.props.annotation;
+    const options = this.getOptions(i);
+    const newIndex = options.indexOf(newOption);
+
+    if (newIndex === -1) {
+      value[i] = { value: newOption.value, option: false };
+    } else {
+      value[i] = { value: newOption.value, option: true };
+    }
+
+    this.clearRelated(i);
+
+    const newAnnotation = Object.assign({}, this.props.annotation, { value });
+    this.props.onChange(newAnnotation);
+  }
+
+  setDefaultValues() {
+    if (!this.props.annotation.value.length) {
+      const value = this.props.task.selects.map(() => ({ option: false, value: null }));
+      const newAnnotation = Object.assign({}, this.props.annotation, { value });
+      this.props.onChange(newAnnotation);
+    }
+  }
+
+  getOptionsKey(i, key) {
+    const { selects } = this.props.task;
+    const select = selects[i];
+    const [parentSelect] = selects.filter(filterSelect => filterSelect.id === select.condition);
+    const parentIndex = selects.indexOf(parentSelect);
+
+    let optionsKey;
+
+    if (this.props.annotation.value[parentIndex] && this.props.annotation.value[parentIndex].option) {
+      optionsKey = this.props.annotation.value[parentIndex].value;
+      if (key) {
+        optionsKey = optionsKey.concat(`;${key}`);
+      }
+    }
+
+    if (parentSelect.condition) {
+      return this.getOptionsKey(parentIndex, optionsKey);
+    }
+    return optionsKey;
+  }
+
+  getOptions(i) {
+    const select = this.props.task.selects[i];
+    const optionsKey = select.condition ? this.getOptionsKey(i) : '*';
+    const options = select.options[optionsKey];
+    return options || [];
+  }
+
+  getDisabledAttribute(i) {
+    const { selects } = this.props.task;
```

Comment:
This function is a little hard to follow. Does it need a comment to clarify what the conditions are for a menu being disabled?